index.js

```diff
@@ -208,6 +208,11 @@ export default function normalizeUrl(urlString, options) {
 		urlObject.searchParams.sort();
 	}
 
+	// Decode query parameters
+	try {
+		urlObject.search = decodeURIComponent(urlObject.search);
```

Comment:
Since the search components are unencoded when `.sort()` is not called, this will potentially decode encoded character where the encoding is meant to be preserved. You could potentially only decode when `!options.sortQueryParameters`, but I think the encoding inconsistency is a bug in the URL implementation and it would be risky to depend on it staying like that.